content/how-to/bpmn/debug-workflows.md

```diff
@@ -0,0 +1,248 @@
+because no `$err` field exists.
+
+{{< tabs >}}
+{{% tab  "Expression" %}}
+```json
+=( 
+  $name := "Rhize";
+
+
+  { 
+  "name": $name,
+  "error": $err
+  }
+)
+```
+{{% /tab %}}
+{{% tab  "Output" %}}
+```
+{
+  "name": "Rhize"
+}
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+You can use this behavior to direct flows.
+For example, an exclusive gateway may have a condition such as `$exists(err)` that flows into an error-handling condition.
+
+### Define error boundary
+
+The error boundary is an intermediate event that evaluates whether a task had an error.
+To use an error boundary:
+1. Attach an intermediate event to a task.
+1. Select the wrench icon and choose the error boundary.
+1. Attach an arrow to the event that directs flows if the node returns an error.
+
+For example, this error boundary directs the control flow when a REST task returns an unexpected result.
+
+{{< bigFigure
+src="/images/bpmn/screenshot-rhize-bpmn-error-boundary.png"
+alt="Screenshot of an error boundary that happens when a REST task returns a non-200 response."
+caption="An error boundary that happens when a REST task returns a non-200 response."
+width="35%"
+>}}
+
+### Create event logging
+
+To persist error handling, you can set gateways that flow to mutation tasks that use the `addEvent` operation. 
+The added event may be a successful operation, an error, or both,
+creating a record of events emitted in the workflow that are stored in your manufacturing knowledge graph.
+This strategy increases the observability of errors and facilitates future analysis.
+It may also be useful when combined with the debugging strategies described in the next section.
+
+## Strategies to debug
+
+For detailed debugging,
+you can use an embedded instance of [Grafana Tempo](https://github.com/grafana/tempo) to inspect each step of the workflow, node by node.
+To debug on the fly, you may also find it useful to use `customResponse` and intermediate message throws to print variables and output at different checkpoints. 
+
+### Debug from the API calls
+
+When you first test or run a workflow, consider starting the testing and debugging process from an API trigger.
+All API triggers return information about the workflow state (for example `COMPLETED` or `ABORTED`).
+With the `createAndRunBpmnSync` operation, you can also use the `customResponse` to provide information from the workflow's variable context.
+For details of how this works, read the guide to [triggering workflows]({{< relref "trigger-workflows" >}}).
+
+For example, consider a workflow that has two nodes, a Message throw event and a REST task.
+1. When the message completes, the user writes `Message sent` into `customResponse` as an output variable.
+1. When the REST task completes, the response is saved into `customResponse`. 
+
+So the `jobState` property reports on the overall workflow status, and `customResponse` serves as a checkpoint to report the state of each node execution.
```

Comment:
For `createAndRunBPMNSync` you can also ask for the `DataJSON` attribute and get the entire variable context at the last node.